src/main/java/org/codehaus/mojo/license/xray/ComponentInfo.java

```diff
@@ -12,7 +13,7 @@ public class ComponentInfo {
     private List<ComponentData> data;
 
     public List<ComponentData> getData() {
-        return data;
+        return data == null ? Collections.emptyList() : data;
```

Comment:
Getter should not modify data, it's a bad practice.

Use @JsonSetter(nulls = Nulls.AS_EMPTY) instead.